mm2src/coins/utxo/rpc_clients.rs

```diff
@@ -2388,32 +2546,445 @@ impl UtxoRpcClientOps for ElectrumClient {
     }
 }
 
+#[async_trait]
+trait ConnMngTrait: Debug {
+    async fn get_conn(&self, address: Option<&str>) -> Option<Arc<AsyncMutex<ElectrumConnection>>>;
+    async fn connect(&self) -> Result<(), String>;
+    async fn is_connected(&self) -> bool;
+    async fn remove_server(&self, address: String) -> Result<(), String>;
+    async fn register(&self, handler: RpcTransportEventHandlerShared);
+    async fn on_verified(&self, address: String);
+    fn on_disconnected(&self, address: String);
+}
+
+#[derive(Debug)]
+struct ConnMngImpl {
+    guarded: AsyncMutex<ConnMngState>,
+    abortable_system: AbortableQueue,
+}
+
+#[derive(Debug)]
+struct ConnMngState {
+    connecting: AtomicBool,
+    event_handlers: Vec<RpcTransportEventHandlerShared>,
+    queue: ConnMngQueue,
+    active: Option<String>,
+    conn_ctxs: BTreeMap<String, ElectrumConnCtx>,
+}
+
+#[derive(Debug)]
+struct ElectrumConnCtx {
+    conn_settings: ElectrumConnSettings,
+    abortable_system: AbortableQueue,
+    suspend_timeout_sec: u64,
+    connection: Option<Arc<AsyncMutex<ElectrumConnection>>>,
+}
+
+#[derive(Clone, Debug)]
+struct ConnMng(Arc<ConnMngImpl>);
+
+impl ConnMng {
+    async fn suspend_server(self, address: String) -> Result<(), String> {
```

Comment:
Since this is a new implementation, please start using specific error types and `MmError`

Reply:
[partially done](https://github.com/KomodoPlatform/komodo-defi-framework/pull/1966/commits/22116e1974c44f061a57fbc10630f4ad879d83f6)